### PR TITLE
Update metrics.py to fix recall calculation

### PR DESCRIPTION
**Pull request by harsha-simhadri**

### CODE REVIEW

**Comment by sourcesync** [comment]
Alright @harsha-simhadri , so after using the new recall, I found that all the databases increased in recall, some suspiciously so.  I looked into it more and I think the fix I suggested has some unintended consequences.  I've added some unit tests to demonstrate here.  The last unit test ( which takes actual data from the bigann dataset ) will fail.

https://github.com/harsha-simhadri/big-ann-benchmarks/blob/gw/T3/t3/tests.py

The edge case that the patch addressed can be addressed better I think:

https://github.com/harsha-simhadri/big-ann-benchmarks/blob/46e0a5f12c420ca5f90f8713abfb964999c2f741/benchmark/plotting/metrics.py#L14-L17

But, even this appears to exhibit an issue, for example I now get impossibly high accuracy on text2image.

Long story short, I'll need to look into this further...

**Comment by sourcesync** [comment]
@harsha-simhadri  OK, i wrote some more unit tests to track down the remaining recall-with-ties issues.  I may have some interesting results.  I believe I've found that each dataset DOES have ties for the public query set GT.  I've been assuming that only msspacev-1B has this issue.  If I'm right about this, then this explains why the recall changes for all datasets when considering ties versus not.  So, that's good.  

The change in recall (ties vs not) appears reasonable except for Text2Image.  I looked into that dataset further.

It looks like the GT distance array for Text2Image is not a monotonically increasing array ( but it is for the other datasets. ). This monotonicity appears to be important for the proper calculation of recall (with ties.)  I guess my question is @harsha-simhadri @maumueller - is the GT distance array supposed to be monotonically increasing for all datasets?

Here is link to all the unit tests including the ones related to monotonicity:

https://github.com/harsha-simhadri/big-ann-benchmarks/blob/t3/unit_tests/t3/tests.py
